cognite/client/utils/_session.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from cognite.client.credentials import OAuthClientCertificate
+from cognite.client.data_classes.iam import ClientCredentials
+from cognite.client.exceptions import CogniteAuthError
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+
+
+def create_session_and_return_nonce(
+    client: CogniteClient,
+    client_credentials: dict | ClientCredentials | None = None,
+    api_name: str = "Functions API",
```

Comment:
Since this is invoked from multiple contexts, let's not have a default value for this.